src/test/java/dog/pawbook/logic/parser/PawbookParserTest.java

```diff
@@ -83,12 +85,64 @@ public void parseCommand_list() throws Exception {
 
     @Test
     public void parseCommand_unrecognisedInput_throwsParseException() {
-        assertThrows(ParseException.class, String.format(MESSAGE_INVALID_COMMAND_FORMAT, HelpCommand.MESSAGE_USAGE), ()
-            -> parser.parseCommand(""));
+        assertThrows(ParseException.class, String.format(MESSAGE_INVALID_COMMAND_FORMAT, HelpCommand.MESSAGE_USAGE), (
+        ) -> parser.parseCommand(""));
     }
 
     @Test
     public void parseCommand_unknownCommand_throwsParseException() {
         assertThrows(ParseException.class, MESSAGE_UNKNOWN_COMMAND, () -> parser.parseCommand("unknownCommand"));
     }
+
+    @Test
+    public void generateAddCommand_unknownEntity_throwsParseException() {
+        assertThrows(ParseException.class, MESSAGE_UNKNOWN_ENTITY, (
+        ) -> parser.generateAddCommand("unknownEntity",
+                        "add owner n/John p/123 e/johnd@example.com a/311, Clementi Ave 2, #02-25 t/friends"));
+    }
+
+    @Test
+    public void generateAddCommand_missingEntity_throwsParseException() {
+        assertThrows(ParseException.class, String.format(MESSAGE_INVALID_COMMAND_FORMAT, AddCommand.MESSAGE_USAGE), (
+        ) -> parser.generateAddCommand("",
+                        "add owner n/John p/123 e/johnd@example.com a/311, Clementi Ave 2, #02-25 t/friends"));
+    }
+
+    @Test
+    public void generateAddCommand_unknownArgument_throwsParseException() {
+        assertThrows(ParseException.class, MESSAGE_UNKNOWN_ARGUMENT, (
+        ) -> parser.generateAddCommand("owner", "unknownArgument"));
+    }
```

Comment:
As commented above, this testing should not happen anymore, but should instead be tested through the `parse` method. Also, try to make use of the constants already defined to come up with the command instead of the magic literals.

Reply:
Noted!